File: College/Assignments/ACPL/Python/2.py
def BinarySearch(arr: list[int], target: int) -> int:
    start, end = 0, len(arr) - 1
    while(start<=end):
        mid = start + (end-start)//2
        if(arr[mid] == target):
            return mid
        else:
            if(arr[mid] > target):
                end = mid-1
            else:
                start = mid+1
    return -1

if __name__ == "__main__":
    arr = [11, 20, 25, 37] 
    print(BinarySearch(arr, 11))

File: College/Assignments/ACPL/CPP/2.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

class Solution{
public:
    int BinarySearch(vector<int>& arr, int target){
        int start = 0, end = arr.size() - 1, mid;
        while (start<=end)
        {
            mid = start + (end-start) / 2;
            if(arr[mid] == target) return mid;
            else{
                if(arr[mid] > target) end = mid-1;
                else start = mid + 1;
            }
        }
        
        return -1;
    }
};


int main(){
    Solution sol;
    vector<int> arr = {11, 20, 25, 37};
    cout<<sol.BinarySearch(arr, 11);
    return 0;
}